portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +24,24 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private final ArrayList<String> messages = new ArrayList<String>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Jonathan");
+    messages.add("Something crazy");
+    messages.add("Something simple");
+    messages.add("Something amazing");
+    String json = "{";
```

Comment:
Alternatively, for creating the json, you could use String.format: https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#format(java.lang.String,%20java.lang.Object...)

Example:
String.format("My name is %s", name);

Reply:
That is a good suggestion.

Reply:
Why not using Gson? It would automatically convert your list to a json String, which is a bit more scalable as you add more messages: https://github.com/google/gson

Reply:
I believe that is next step, but I might be wrong.

Reply:
Gson seems like a good way for the code to not just apply to one set messages. I've never used it before though.

Reply:
Once you've added the dependency (which is explained in the week 3 walkthrough), you can do:

```java
private final Gson gson = new Gson();


private void myFunction() {
  String json = gson.toJson(messageList);
}
```

This will convert your array / list to a json String automatically.

Let me know if you need more help, happy to provide more details. This [page](https://www.tutorialspoint.com/gson/gson_quick_guide.htm) might also be useful to see an example on how Gson is setup. Note that they use a builder, which I don't think we have to worry at this time, so you can just instantiate Gson right away.

Reply:
Okay thank you.